webapi/CopilotChat/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -427,10 +427,14 @@ public async Task<string> ExtractAudienceAsync(SKContext context)
 
         // Render the prompt
         var promptRenderer = new PromptTemplateEngine();
-        var renderedPrompt = await promptRenderer.RenderAsync(
-            this._promptOptions.SystemChatPrompt,
-            chatContext);
+        var renderedPrompt = await promptRenderer.RenderAsync(this._promptOptions.SystemChatPrompt, chatContext, cancellationToken);
         chatContext.Variables.Set("prompt", renderedPrompt);
+
+        // Need to extract this from the rendered prompt because Time and Date are calculated during render
+        var systemChatContinuation = Regex.Match(renderedPrompt, @"(SINGLE RESPONSE FROM BOT TO USER:\n\[.*] bot:)").Value;
```

Comment:
nit: this seems a little but hacky. But I also don't have a good solution :(

Reply:
The only other way I can think of doing it is rendering the SystemChatContinuation dynamically like this 
![image](https://github.com/microsoft/chat-copilot/assets/125500434/f25eb990-063c-4901-b220-12dec1d62d86)

![image](https://github.com/microsoft/chat-copilot/assets/125500434/4c0361c5-2720-40b8-944f-0dfcd34d7547)

Reply:
I guess now is fine. Maybe we can save the regex as a const somewhere?

Reply:
Parse from the renderedPrompt using the initial Regex? or dynamically calculating the SystemChatContinuation?

Reply:
Sorry, I meant the pattern.

Reply:
Done added as a const to PromptOptions 
![image](https://github.com/microsoft/chat-copilot/assets/125500434/fdd13092-742f-46b4-a127-00eced72b8d8)